backend/internal/handlers/friendship/friendship.go

```diff
@@ -1,28 +1,43 @@
 package friendship
 
 import (
-	"fmt"
 	"github.com/GenerateNU/nightlife/internal/models"
 	"github.com/gofiber/fiber/v2"
-	"log"
+	"github.com/google/uuid"
 )
 
 // POST endpoint to create friendship between two users in DB
 func (s *Service) CreateFriendship(c *fiber.Ctx) error {
-	fmt.Println("Creating a friendship")
+    var req models.Friendship
 
-	var req models.Friendship
+    // Parse the request body and check for errors
+    if err := c.BodyParser(&req); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
```

Comment:
would recommend using [`errs.InvalidJSON(err)`](https://github.com/GenerateNU/nightlife/blob/main/backend/internal/errs/api_err.go#L32). would refactor that func to be
```go
func InvalidJSON(err error) APIError {
	return NewAPIError(http.StatusBadRequest, err)
}
```
so we can drill down the actual error to the client. more ergonomic since we can tell them they forgot to close with a `}` instead of a blanket `Cannot parse JSON`